jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -324,6 +318,16 @@ public static boolean isValidJavaClass(String className) {
     return true;
   }
 
+  /**
+   * @param list the list to copy
+   * @return an {@code ImmutableList} containing all non-null, non-empty elements of the original
+   *     list
+   */
+  public static ImmutableList<String> filterNullOrEmpty(List<String> list) {
```

Comment:
This method should probably just go into the Maven plugin since this error was only due to Maven and we probably don't want to be applying this filtering to Gradle since it is not visible to the user and we may be covering up bad user configuration this way.

Reply:
In that case should we should just be filtering on null, not nullOrEmpty?

Reply:
Yea, that might be good, since an empty element should be fine - it just won't add anything to the entrypoint. For the empty case, we could provide a warning for possibly unintended configuration, but that could be a small fix for another PR.

Reply:
This error does still happen with gradle if a user intentionally adds a null element to the array, so I think whatever we do, we should handle the error in both plugins. Would it make sense to treat all cases the same, and warn about + filter both empty and null elements in both plugins? And if so, what class should the warning+filtering be going in?

Reply:
I think for Gradle, if the user intentionally adds a null element to the array, then they intentionally made a bad configuration. However, in Maven, it is more of a bug, where the user did not intentionally add null but it generated a null. In general, I think it should be like:

1. If user made a bad configuration intentionally, it should error.
1. If the bad configuration was unintentional, we should error/warn/or fix it automatically.

For where the warning+filtering should go, it should definitely be a frontend class, or if it is specific to Maven/Gradle, it should go in the respective maven/gradle plugin.